src/main/java/br/com/apollomusic/app/model/services/EstablishmentService.java

```diff
@@ -0,0 +1,36 @@
+package br.com.apollomusic.app.model.services;
+
+import br.com.apollomusic.app.model.dto.ErrorResDto;
+import br.com.apollomusic.app.model.entities.Establishment;
+import br.com.apollomusic.app.repository.EstablishmentRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class EstablishmentService {
+    private final EstablishmentRepository establishmentRepository;
+
+    public EstablishmentService(EstablishmentRepository establishmentRepository) {
+        this.establishmentRepository = establishmentRepository;
+    }
+
+    public ResponseEntity<?> turnOn(Long establishmentId){
+        Establishment establishment = establishmentRepository.findById(establishmentId).get();
+        if (establishment.isOff()){
+            establishment.setOff(false);
+            establishmentRepository.save(establishment);
+            return ResponseEntity.ok(establishment);
+        }
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Erro ao ligar Estabelicimento");
```

Comment:
Ambas as funções retornam "NOT FUND" mesmo que tenha ocorrido outro problema.

Sugestão:
Adicione uma verificação se establishment possui algum valor, isso por que caso não possuir significa que não foi encontrado.

if(establishment.isEmpty()) ...

Nesse cenário caso for vazio vc retorne um erro NOT FOUND.

Tecnicamente o código funcionará igual, porém o frontend não conhece o backend, se vc me retornar apenas NOT FOUND no frontend será exibido que não foi encontrado.

O comentário serve para as duas funções.

Reply:
Tentei desativar um id inexistente, a rota me retorna erro 500, esse tipo de erro bloqueia operação no front.

Nesse cenário era para ter me retorando erro 404 NOT FOUND

<img width="1021" alt="Screenshot 2024-06-27 at 08 15 39" src="https://github.com/caioslopes/backend-apollo/assets/99999453/2411f463-c422-4d10-9e33-78f5e79e6c4e">